LEGO1/mxpresenter.h

```diff
@@ -68,17 +68,19 @@ class MxPresenter : public MxCore
 
   MxBool IsEnabled();
 
-  inline MxS32 GetCurrentTickleState() { return this->m_currentTickleState; }
-  inline MxPoint32 GetLocation() { return this->m_location; }
-  inline MxS32 GetDisplayZ() { return this->m_displayZ; }
-  inline MxDSAction *GetAction() { return this->m_action; }
+  inline MxS32 GetCurrentTickleState() const { return this->m_currentTickleState; }
+  inline MxPoint32 GetLocation() const { return this->m_location; }
+  inline MxS32 GetDisplayX() const { return this->m_location.m_x; }
```

Comment:
I'd call this `GetLocationX` (same with Y) since it appears to be somewhat distinct from the Z value (also confirmed by the 1996 source).

Did you try adding getters like `GetX` (and Y) to `MxPoint32` such that the call would be `GetLocation().GetX()` or does this also lower accuracy? If not, I think that might be the cleanest. I've seen those getters in the 1996 source on that class.

Reply:
I just tried with GetX() and GetY() and it's the same as using `m_location.m_x` directly. With that you get an `LEA` on the address of the MxPoint32 member. To illustrate what I mean:

```assembly
; ECX points to m_waitIndicator:
  00044 8d 71 10    lea esi, DWORD PTR [ecx+16]

; [...]
; 361  :     m_copyRect.left = m_waitIndicator->GetLocation().GetX();

  0004b 8b 06       mov eax, DWORD PTR [esi]
  0004d 89 43 0c    mov DWORD PTR [ebx+12], eax

; 362  :     m_copyRect.top = m_waitIndicator->GetLocation().GetY();

  00050 8b 46 04    mov eax, DWORD PTR [esi+4]
  00053 89 43 10    mov DWORD PTR [ebx+16], eax
```

Versus the original code that just grabs the values as if there were no struct at all:

```assembly
; ECX points to m_waitIndicator:
1004c5ba 8b 79 10   MOV     EDI,dword ptr [ECX + 0x10] ; m_location.X
1004c5bd 8b 46 54   MOV     EAX,dword ptr [ESI + 0x54]
1004c5c0 45         INC     EBP
1004c5c1 c1 e8 03   SHR     EAX,0x3
1004c5c4 8b 51 14   MOV     EDX,dword ptr [ECX + 0x14] ; m_location.Y
```

So I'll change it to GetLocationX() and Y. Is the Z value in this context meant to be the z-depth (or z-index in HTML land)? As in: we have screen x and y and an order to draw things, not a full 3d coodinate? (Maybe that's just splitting hairs, don't mind me)

Last thing: I changed the malloc/free in TransitionManager to new/delete to match what we're doing everywhere else.

Reply:
> Is the Z value in this context meant to be the z-depth (or z-index in HTML land)? As in: we have screen x and y and an order to draw things, not a full 3d coodinate? (Maybe that's just splitting hairs, don't mind me)

Something like that I'd say, yes. I don't know for sure yet but considering they chose to separate the Z-value into its own member separate from the X/Y point with a distinct name (`displayZ` - that's how they call it in the 1996 source). The presenters are also sorted according to their Z-value here: https://github.com/isledecomp/isle/blob/master/LEGO1/mxvideomanager.cpp#L76